app/helpers/insured/group_selection_helper.rb

```diff
@@ -328,61 +328,32 @@ def shop_health_and_dental_relationship_benefits(employee_role, benefit_group)
-      end
-      true if found
-    end
-
-    def is_person_with_broker_role?(person, hbx_enrollment)
-      return true if person.broker_role&.broker_agency_profile && hbx_enrollment.broker_agency_profile_id.present? && person.broker_role.broker_agency_profile.id == hbx_enrollment.broker_agency_profile_id
-      return unless person.broker_role && hbx_enrollment.broker.present?
-      true if person.broker_role.id == hbx_enrollment.broker&.id
-    end
-
-    def is_general_agency_authorized?(user, hbx_enrollment_id)
-      person = user.person
-      return unless person.present?
-      staff_roles = person.active_general_agency_staff_roles
-      return unless staff_roles.present?
-      general_agency_profile_id = staff_roles.last.benefit_sponsors_general_agency_profile_id
-      organization = BenefitSponsors::Organizations::Organization.where(
-        "profiles.default_general_agency_profile_id" => BSON::ObjectId.from_string(general_agency_profile_id.to_s)
-      ).last
-      hbx_enrollment = HbxEnrollment.where(id: hbx_enrollment_id).last
-      return unless organization.present? && hbx_enrollment.present?
-      broker_agency_profile = organization.broker_agency_profile
-      broker_ids = fetch_brokers_from_enrollment(hbx_enrollment)
-      true if broker_agency_profile && broker_ids.include?(broker_agency_profile.primary_broker_role.id)
-    end
-
-    # A person (enrollment holder) may serve as an employee for two different employers,
-    # and each employer may have different brokers assigned to the person.
-    # Additionally, a person can have a dual role, meaning they can hold both employee and consumer roles.
-    #
-    # Furthermore, there are a few other cases that can occur:
-    # If a person adds a broker and purchases an enrollment, the broker is assigned to the hbx_enrollment
-    # If a person purchases an enrollment and then adds a broker, the broker is not assigned to the hbx_enrollment
-    # In both the above cases broker have full control on the person.
-    def fetch_brokers_from_enrollment(hbx_enrollment)
-      return [hbx_enrollment.broker.id] if hbx_enrollment.broker.present?
-      broker_ids = []
-      primary_person = hbx_enrollment.family&.primary_person
-      return broker_ids unless primary_person
-      broker_ids << primary_person.fetch_writing_agents_for_employee_role
-      broker_ids << primary_person.primary_family&.current_broker_agency&.writing_agent&.id if primary_person.has_consumer_role?
-      broker_ids.compact.flatten
+    def is_broker_authorized?(current_user, family)
+      person = current_user.person
+      return false if person.blank?
+
+      logged_user_broker_role = person.broker_role
+      logged_user_staff_roles = person.broker_agency_staff_roles.where(aasm_state: 'active')
+      return false if logged_user_broker_role.blank? && logged_user_staff_roles.blank? # logged in user is not a broker
```

Comment:
This condition will block users with only broker staff roles (Users that are not primary broker). Please let me know if it's expected?

Reply:
no, it returns `false` only if user doesn't have any of broker role or broker staff roles.